django/utils/timesince.py

```diff
@@ -40,10 +40,15 @@ def timesince(d, now=None, reversed=False):
     if not now:
         now = datetime.datetime.now(utc if is_aware(d) else None)
 
-    delta = (d - now) if reversed else (now - d)
+    if reversed:
+        d, now = now, d
+    delta = now - d
 
     # Deal with leapyears by subtracing the number of leapdays
-    delta -= datetime.timedelta(calendar.leapdays(d.year, now.year))
+    leapdays = calendar.leapdays(d.year, now.year)
+    if leapdays != 0 and calendar.isleap(d.year) and d.date() > datetime.date(d.year, 2, 29):
```

Comment:
`timesince` and `timeuntil` returns result up to two adjacent units, hence we can always subtract this one leap day and result will be this same, e.g. `365 -> 1 year`, `366 -> 1 year`. I suppose that we can omit this condition and change release notes to `Fixed a regression in the ``timesince`` and ``timeuntil`` filters that caused incorrect results for dates in a leap year.`